cirq-core/cirq/circuits/qasm_output.py

```diff
@@ -289,22 +289,25 @@ def output(text):
                 output(f'qubit[{len(self.qubits)}] q;\n')
         # Classical registers
         # Pick an id for the creg that will store each measurement
-        already_output_keys: Set[str] = set()
+        # cregs will store key -> (#qubits, comment)
+        cregs: Dict[str, tuple[int, str]] = {}
```

Comment:
If you did this in the QasmOutput constructor and stored it in QasmArgs, you could pass it into [KeyCondition.qasm](https://github.com/quantumlib/Cirq/blob/b35382fef518718608a5ca152a576e0fb705186a/cirq-core/cirq/value/condition.py#L116), and fix #5691.

The reason it's required is that KeyCondition is triggered if `meas[m_key] != 0`. In OpenQasm2, the only logical comparator is equality, so `!= 0` can only be represented if the creg is a single bit (i.e. it would convert the condition to `if m_key == 1`). If the creg is multiple bits, then `if m_key == 1` would be incorrect, and `m_key != 0` is not representable. So that's why the existing behavior is to play it safe and just fail.

Going a little bit further, in OpenQasm3, it looks like `m_key != 0` *is* representable, so checking `QasmArgs.version` in KeyCondition.qasm would allow supporting multi-bit classical controls if `version == '3.0'`. OpenQasm3 could also potentially support serializing additional types of logic in SympyConditions.

Reply:
It might also be useful to have this be a function on `Circuit` or part of the `measurement_keys` protocol. I'd imagine "get the shapes of all the measurement keys in the circuit" would be a common need.

Reply:
Changed to put into constructor, but I am not convinced that helps, since we don't pass QasmOutput during the qasm protocol.   In any case, I will deal with classical controls in a follow-up PR.

Also, Circuit has enough cruft on it that I don't think it's a good idea to add more functions onto it, so I will leave it here.